PrimeNumberTheoremAnd/MellinCalculus.lean

```diff
@@ -117,6 +117,49 @@ Almost by definition.
 \end{proof}
 %%-/
 
+-- TODO: upstream to mathlib Arctan.lean
+lemma arctan_atTop : Tendsto arctan atTop (𝓝 (π / 2)) :=
+  tendsto_nhds_of_tendsto_nhdsWithin (tendsto_Ioo_atTop.mp tanOrderIso.symm.tendsto_atTop)
+
+lemma arctan_atBot : Tendsto arctan atBot (𝓝 (-(π / 2))) :=
+  tendsto_nhds_of_tendsto_nhdsWithin (tendsto_Ioo_atBot.mp tanOrderIso.symm.tendsto_atBot)
+
+lemma arctan_ne_zero {x : ℝ} (hx : x ≠ 0) : arctan x ≠ 0 :=
+  fun h ↦ hx <| (show arctan.Injective from StrictMono.injective tanOrderIso.symm.strictMono)
+    (h.trans arctan_zero.symm)
+
+-- TODO: upstream to mathlib ImproperIntegral.lean
+private lemma intervalIntegral_one_div_one_add_sq_tendsto :
+    Tendsto (fun i => ∫ (x : ℝ) in -i..i, 1 / (1 + x ^ 2)) atTop (𝓝 π) := by
+  convert Tendsto.add arctan_atTop arctan_atTop <;> simp
+
+open MeasureTheory in
+lemma integrable_one_div_one_add_sq : Integrable fun (x : ℝ) ↦ 1 / (1 + x ^ 2) := by
+  have (x : ℝ) : ‖1 / (1 + x ^ 2)‖ = 1 / (1 + x ^ 2) := norm_of_nonneg (by positivity)
+  refine integrable_of_intervalIntegral_norm_tendsto π (fun i ↦ ?_) tendsto_neg_atTop_atBot
+    tendsto_id (by simpa only [this] using intervalIntegral_one_div_one_add_sq_tendsto)
+  cases eq_zero_or_neZero i with
```

Comment:
Does the `NeZero` actually help or can this be `by_cases hi : i = 0`?

Reply:
that works better, thanks!